test/ScanDependencies/wrong_target_candidate_diagnostic.swift

```diff
@@ -0,0 +1,16 @@
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t/module-cache)
+// RUN: %empty-directory(%t/moduleInputs)
+// RUN: split-file %s %t
+
+// ERROR: error: unable to resolve Swift module dependency to a compatible module: 'FooBar'
+// ERROR: note: found incompatible module '{{.*}}{{/|\\}}moduleInputs{{/|\\}}FooBar.swiftmodule': compiled for a different target platform
+
+// RUN: %target-swift-frontend -emit-module %t/FooBar.swift -emit-module-path %t/moduleInputs/FooBar.swiftmodule -module-name FooBar -target x86_64-unknown-linux-gnu
```

Comment:
Does this test build? I will expect almost no build produce stdlib for both linux-gnu and macosx. Maybe you can get away with `-parse-stdlib`?